### ISSUE TITLE
Parallels: hitting dpkg issue again (see desc)

### ISSUE THREAD

**Issue by wizonesolutions** [chef/bento#695]
Seems to be the same as https://github.com/chef/bento/issues/609

Using `bento/ubuntu-16.04` version `2.3.0`. When I try to run `apt-get` during provisioning, I am always met with:

```
fatal: [default]: FAILED! => {"changed": false, "cmd": "apt-get update", "failed": true, "msg": "E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)\nE: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?", "rc": 100, "stderr": "E: Could not get lock /var/lib/dpkg/lock - open (11: Resource temporarily unavailable)\nE: Unable to lock the administration directory (/var/lib/dpkg/), is another process using it?\n", "stdout": "Hit:1 http://security.ubuntu.com/ubuntu xenial-security InRelease\nHit:2 http://us.archive.ubuntu.com/ubuntu xenial InRelease\nHit:3 http://us.archive.ubuntu.com/ubuntu xenial-updates InRelease\nHit:4 http://us.archive.ubuntu.com/ubuntu xenial-backports InRelease\n", "stdout_lines": ["Hit:1 http://security.ubuntu.com/ubuntu xenial-security InRelease", "Hit:2 http://us.archive.ubuntu.com/ubuntu xenial InRelease", "Hit:3 http://us.archive.ubuntu.com/ubuntu xenial-updates InRelease", "Hit:4 http://us.archive.ubuntu.com/ubuntu xenial-backports InRelease"]}
```

Let me know what info I can provide to help reproduce.

**Comment by rickard-von-essen** [comment]
Which is the other process locking apt?

**Comment by wizonesolutions** [comment]
Hmm. I don't think it is `unattended-upgrades`. @legal90 does the latest `vagrant-parallels` or Parallels itself or maybe the Bento machines attempt to background-update the Tools? This output is not clear, because it's not formatted well, but it kind of looks like `apt-get` was kicked off by a `prl*` process?

````
==> default: Running provisioner: shell...
    default: Running: inline script
==> default: mesg:
==> default: ttyname failed
==> default: :
==> default: Inappropriate ioctl for device
==> default: This is what is locking dpkg:
==> default: COMMAND PID USER FD TYPE DEVICE SIZE/OFF NODE NAME apt-get 2745 root 8uW REG 252,0 0 131744 /var/lib/dpkg/lock
==> default: The tree:
==> default: /var/lib/dpkg/lock:
==> default: systemd-+-accounts-daemon-+-{gdbus} | `-{gmain} |-acpid |-agetty |-atd |-cron |-dbus-daemon |-dhclient |-irqbalance |-2*[iscsid] |-lvmetad |-lxcfs---2*[{lxcfs}] |-mdadm |-ondemand---sleep |-polkitd-+-{gdbus} | `-{gmain} |-prltoolsd-+-prltimesync | |-4*[prltoolsd] | `-6*[{prltoolsd}] |-ptiagent-cmd-+-install---install-kmods.s---pm.sh---apt-get---http | |-{CInstallationLo} | `-{QProcessManager} |-rpc.statd |-rpcbind |-rsyslogd-+-{in:imklog} | |-{in:imuxsock} | `-{rs:main Q:Reg} |-snapd---7*[{snapd}] |-sshd---sshd---sshd---sudo---bash---bash---bash-+-pstree | `-sudo---fuser |-systemd---(sd-pam) |-systemd-fsckd |-systemd-hostnam |-systemd-journal |-systemd-logind |-systemd-timesyn---{sd-resolve} `-systemd-udevd
````

The script was an inline script that ran this:

````bash
echo This is what is locking dpkg:
echo `sudo lsof /var/lib/dpkg/lock`
echo 'The tree: '
echo `sudo fuser /var/lib/dpkg/lock | pstree`
````

**Comment by wizonesolutions** [comment]
Here we go. I fixed the output:

````
==> default: This is what is locking dpkg:
==> default: COMMAND  PID USER   FD   TYPE DEVICE SIZE/OFF   NODE NAME
==> default: apt-get 2750 root    9uW  REG  252,0        0 131744 /var/lib/dpkg/lock
==> default: The tree:
==> default: /var/lib/dpkg/lock:
==> default: systemd-+-accounts-daemon-+-{gdbus}
==> default:         |                 `-{gmain}
==> default:         |-acpid
==> default:         |-agetty
==> default:         |-atd
==> default:         |-cron
==> default:         |-dbus-daemon
==> default:         |-dhclient
==> default:         |-irqbalance
==> default:         |-2*[iscsid]
==> default:         |-lvmetad
==> default:         |-lxcfs---2*[{lxcfs}]
==> default:         |-mdadm
==> default:         |-ondemand---sleep
==> default:         |-polkitd-+-{gdbus}
==> default:         |         `-{gmain}
==> default:         |-prltoolsd-+-prltimesync
==> default:         |           |-2*[prltoolsd]
==> default:         |           `-6*[{prltoolsd}]
==> default:         |-ptiagent-cmd-+-install---install-kmods.s---pm.sh---apt-get---http
==> default:         |              |-{CInstallationLo}
==> default:         |              `-{QProcessManager}
==> default:         |-rpc.statd
==> default:         |-rpcbind
==> default:         |-rsyslogd-+-{in:imklog}
==> default:         |          |-{in:imuxsock}
==> default:         |          `-{rs:main Q:Reg}
==> default:         |-snapd---6*[{snapd}]
==> default:         |-sshd---sshd---sshd---sudo---bash---bash---bash-+-pstree
==> default:         |                                                `-sudo
==> default:         |-systemd---(sd-pam)
==> default:         |-systemd-fsckd
==> default:         |-systemd-hostnam
==> default:         |-systemd-journal
==> default:         |-systemd-logind
==> default:         |-systemd-timesyn---{sd-resolve}
==> default:         `-systemd-udevd
````

So then it's Parallels. Perhaps I can configure the option to update the Tools during startup instead.

**Comment by wizonesolutions** [comment]
Yep. Configurable.

https://parallels.github.io/vagrant-parallels/docs/configuration.html

````ruby
config.vm.provider "parallels" do |prl|
  prl.check_guest_tools = false
end
````

Guess I will just set that since I don't want to wait 2-6 mins, and the tools in the Bento box are probably sufficiently up-to-date that I don't have to care.

Closing this, since it's Parallels' fault, and there is a workaround.

**Comment by wizonesolutions** [comment]
Hrm. Configuring that DID get rid of the message, but it didn't actually stop `apt-get` from being run upon the box booting. I have a workaround now (I wait for `apt-get` to not be running before I start provisioning), but I don't think this should be turned on by default in base boxes because the first thing most of us do in provisioning scripts is update packages!

**Comment by cblecker** [comment]
I started hitting this same issue with `bento/ubuntu-14.04` at box version `2.3.0`. Reverting the box to `2.2.9` goes back to the expected behavior.

**Comment by rickard-von-essen** [comment]
Ping @legal90

**Comment by legal90** [comment]
Yeah, I can confirm that in Parallels Tools >= 12.0.0 (present in bento boxes v2.3.0+) there is a mechanism trying to update Parallels Tools in the background right after the VM boot. It is done by executing `ptiagent-cmd`, which causes `apt-get` commands run.

That might be not a great decision since it causes such issues with Ubuntu guests like this one. We are discussing it at Parallels and hopefully this behaviour will be fixed in the one of upcoming updates of Parallels Desktop.

@wizonesolutions Please, try to apply this trick to your Vagrantfile. It should turn off that internal updater.
```
  config.vm.provider "parallels" do |prl|
    prl.customize ["set", :id, "--tools-autoupdate", "no"]
  end
```
Does it help?

**Comment by cblecker** [comment]
If this works, is it somehow possible to build that option in at install time, until such time as the Tools default behaviour itself can be looked at?
(https://github.com/chef/bento/blob/master/scripts/common/parallels.sh)

**Comment by wizonesolutions** [comment]
This worked, by the way. My "check if still updating" script finished immediately when I put that in.

**Comment by legal90** [comment]
@wizonesolutions This issue should be fixed in vagrant-parallels [v1.7.1](https://github.com/Parallels/vagrant-parallels/releases/tag/v1.7.1). 
We've turned off `--tools-autoupdate` there by default. 

It might be also reasonable to disable it in Packer too, as @cblecker has suggested above.
